lib/spack/spack/util/gcs.py

```diff
@@ -0,0 +1,216 @@
+
+        Uses GCS Batch operations to bundle several delete operations together.
+        """
+        from google.cloud.exceptions import NotFound
+        tty.debug("Bucket.destroy(recursive={0})".format(recursive))
+        try:
+            bucket_blobs = self.get_all_blobs(recursive=recursive, relative=False)
+            batch_size = 1000
+
+            num_blobs = len(bucket_blobs)
+            for i in range(0, num_blobs, batch_size):
+                with self.client.batch():
+                    for j in range(i, min(i + batch_size, num_blobs)):
+                        blob = self.blob(bucket_blobs[j])
+                        blob.delete()
+        except NotFound as ex:
+            tty.error("{0}, Could not delete a blob in bucket {1}.".format(
+                      ex, self.name))
+            sys.exit(1)
+
+
+class GCSBlob(object):
+    """GCS Blob object
+
+    Wraps some blob methods for spack functionality
+    """
+    def __init__(self, url, client=None):
+
+        self.url = url
+        if url.scheme != 'gs':
+            raise ValueError('Can not create GCS blob connection with scheme: {SCHEME}'
+                             .format(SCHEME=url.scheme))
+
+        if not client:
+            self.client = gcs_client()
+        else:
+            self.client = client
+
+        self.bucket = GCSBucket(url)
+
+        self.blob_path = self.url.path.lstrip('/')
+
+        tty.debug("New GCSBlob")
+        tty.debug("  blob_path = {0}".format(self.blob_path))
+
+        if not self.bucket.exists():
+            tty.warn("The bucket {0} does not exist, it will be created"
+                     .format(self.bucket.name))
+            self.bucket.create()
+
+    def get(self):
+        return self.bucket.get_blob(self.blob_path)
+
+    def exists(self):
+        from google.cloud.exceptions import NotFound
+        try:
+            blob = self.bucket.blob(self.blob_path)
+            exists = blob.exists()
+        except NotFound:
+            return False
+
+        return exists
+
+    def delete_blob(self):
+        from google.cloud.exceptions import NotFound
+        try:
+            blob = self.bucket.blob(self.blob_path)
+            blob.delete()
+        except NotFound as ex:
+            tty.error("{0}, Could not delete gcs blob {1}".format(ex, self.blob_path))
+
+    def upload_to_blob(self, local_file_path):
+        blob = self.bucket.blob(self.blob_path)
+        blob.upload_from_filename(local_file_path)
+
+    def get_blob_byte_stream(self):
+        return self.bucket.get_blob(self.blob_path).open(mode='rb')
+
+    def get_blob_headers(self):
+        blob = self.bucket.get_blob(self.blob_path)
+
+        headers = {}
+        headers['Content-type'] = blob.content_type
+        headers['Content-encoding'] = blob.content_encoding
+        headers['Content-language'] = blob.content_language
+        headers['MD5Hash'] = blob.md5_hash
```

Comment:
Here we can use a dict literal:
```suggestion
        headers = {
            'Content-type': blob.content_type,
            'Content-encoding': blob.content_encoding,
            'Content-language': blob.content_language,
            'MD5Hash': blob.md5_hash
        }
```

Reply:
👍🏻